prov/gni/include/gnix_cq.h

```diff
@@ -45,6 +45,9 @@
 #define GNIX_CQ_DEFAULT_FORMAT struct fi_cq_entry
 #define GNIX_CQ_DEFAULT_SIZE   256
 
+/* forward declaration */
+struct gnix_fid_ep;
```

Comment:
Yes, I added `#include "gnix.h"` to gnix_cq.h and removed `#include "gnix.h"` from gnix_cq.c.